File: plugins/DataStreamOpenOCD/gdb/GdbVar.java
/*    */ package com.st.stmstudio.gdb;
/*    */ 
/*    */ public class GdbVar
/*    */ {
/*    */   private String m_name;
/*    */   private short m_type;
/*    */   private String m_filename;
/*    */   private String m_displayFilename;
/*    */   private long m_address;
/*    */ 
/*    */   GdbVar(String paramString1, String paramString2)
/*    */   {
/* 11 */     this.m_name = paramString1;
/* 12 */     this.m_filename = paramString2;
/* 13 */     ParseFileDisplayName();
/* 14 */     this.m_type = -1;
/* 15 */     this.m_address = 0L;
/*    */   }
/*    */ 
/*    */   GdbVar(String paramString1, short paramShort, String paramString2, long paramLong) {
/* 19 */     this.m_name = paramString1;
/* 20 */     this.m_filename = paramString2;
/* 21 */     ParseFileDisplayName();
/* 22 */     this.m_type = paramShort;
/* 23 */     this.m_address = paramLong;
/*    */   }
/*    */ 
/*    */   private void ParseFileDisplayName() {
/* 27 */     int i = this.m_filename.lastIndexOf("\\");
/* 28 */     if (i != -1) {
/* 29 */       String str = this.m_filename.substring(0, i);
/* 30 */       this.m_displayFilename = (this.m_filename.substring(i + 1) + " (" + str + ")");
/*    */     } else {
/* 32 */       this.m_displayFilename = this.m_filename;
/*    */     }
/*    */   }
/*    */ 
/*    */   public String get_name() {
/* 37 */     return this.m_name;
/*    */   }
/*    */ 
/*    */   public short get_type() {
/* 41 */     return this.m_type;
/*    */   }
/*    */ 
/*    */   public String get_filename() {
/* 45 */     return this.m_displayFilename;
/*    */   }
/*    */ 
/*    */   public long get_address() {
/* 49 */     return this.m_address;
/*    */   }
/*    */ 
/*    */   public void set_address(long paramLong) {
/* 53 */     this.m_address = paramLong;
/*    */   }
/*    */ }

/* Location:           /home/mm/STMStudio.jar
 * Qualified Name:     com.st.stmstudio.gdb.GdbVar
 * JD-Core Version:    0.6.2
 */

File: plugins/DataStreamOpenOCD/gdbvar.cpp
#include "gdbvar.h"

#include <stdio.h>
#include <QDebug>

GdbVar::GdbVar(const QString &name, const QString &fileName)
{
  m_name = name;
  m_filename = fileName;
  ParseFileDisplayName();
  m_type = -1;
  m_address = 0L;
}

GdbVar::GdbVar(const QString &name, short paramShort, const QString &fileName, long address)
{
  m_name = name;
  m_filename = fileName;
  ParseFileDisplayName();
  m_type = paramShort;
  m_address = address;
}

void GdbVar::ParseFileDisplayName()
{
  int i = m_filename.lastIndexOf("\\");
  if (i == -1)
    i = m_filename.lastIndexOf("/");

  if (i != -1) {
    QString str = m_filename.left(i);
    m_displayFilename = (m_filename.mid(i + 1) + " (" + str + ")");
  } else {
    m_displayFilename = m_filename;
  }
}

QString GdbVar::get_name() const
{
  return m_name;
}

short GdbVar::get_type() const
{
  return m_type;
}

QString GdbVar::get_filename() const
{
  return m_displayFilename;
}

long GdbVar::get_address()
{
  return m_address;
}

void GdbVar::set_address(long paramLong)
{
  m_address = paramLong;
}

void GdbVar::dump()
{
  printf("%s %s, 0x%lx\n", m_filename.toLocal8Bit().constData(), m_name.toLocal8Bit().constData(), m_address);
  //qWarning() << m_filename << m_name << QString::number(m_address, 16);
}
